js/components/Star/index.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component, PropTypes } from 'react';
+import {
+  TouchableOpacity,
+  Image,
+  StyleSheet,
+} from 'react-native';
+
+export default class Star extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: false,
+    };
+
+    this.toggle = this.toggle.bind(this);
+  }
+
+  toggle() {
+    this.setState({ value: !this.state.value });
+  }
+
+  render() {
+    const img = this.state.value ?
+      <Image source={require('./assets/star.png')}/> :
+      <Image source={require('./assets/star-border.png')}/>;
+
+    return (
+      <TouchableOpacity
+        style={styles.container}
+        onPress={this.toggle}>
```

Comment:
I spoke with @arcturus about an idea for component design. We can have a dumb component that is just a `<Star selected={true}/> that toggles between `true` and `false`. But then we can wrap that in a `<MagnetStar itemId={...}>` component that wires the component into the app's Redux store.

This means that we could drop a `<MagnetStar>` anywhere in the app and it'll "just work".